leakcanary-android-core/src/main/java/leakcanary/internal/HeapDumpPolicy.kt

```diff
@@ -0,0 +1,28 @@
+package leakcanary.internal
+
+import leakcanary.LeakCanary
+import leakcanary.internal.HeapDumpPolicy.HeapDumpStatus.DISABLED_CONFIG_CHANGED
+import leakcanary.internal.HeapDumpPolicy.HeapDumpStatus.DISABLED_DEBUGGER_ATTACHED
+import leakcanary.internal.HeapDumpPolicy.HeapDumpStatus.ENABLED
+
+object HeapDumpPolicy {
+  enum class HeapDumpStatus {
+    ENABLED,
+    DISABLED_DEBUGGER_ATTACHED,
+    DISABLED_CONFIG_CHANGED
+  }
+
+  fun getHeapDumpStatus(): HeapDumpStatus {
+    val config = LeakCanary.config
+
+    if (!config.dumpHeapWhenDebugging && DebuggerControl.isDebuggerAttached) {
+      return DISABLED_DEBUGGER_ATTACHED
+    }
+
+    if (!config.dumpHeap) {
```

Comment:
Instead of surfacing just the current state, we should surface why we're in that state.

Reply:
I now have 4 enums:

ENABLED, DISABLED_DEBUGGER_ATTACHED, DISABLED_BY_APP, DISABLED_RUNNING_TESTS.

Do they meet all the requirements?